sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -865,29 +993,61 @@ static void *_lwmac_thread(void *args)
                 msg_reply(&msg, &reply);
                 break;
             }
+
             case GNRC_NETAPI_MSG_TYPE_GET: {
                 /* TODO: filter out MAC layer options -> for now forward
                          everything to the device driver */
-                LOG_DEBUG("[LWMAC] GNRC_NETAPI_MSG_TYPE_GET received\n");
+                puts(" GNRC_NETAPI_MSG_TYPE_GET received.");
                 /* read incoming options */
                 opt = (gnrc_netapi_opt_t *)msg.content.ptr;
                 /* get option from device driver */
-                res = dev->driver->get(dev, opt->opt, opt->data, opt->data_len);
-                LOG_DEBUG("[LWMAC] Response of netdev->get: %i\n", res);
+                res = netif->ops->get(netif, opt);
+                LOG_DEBUG("[LWMAC] Response of netif->ops->get(): %i\n", res);
                 /* send reply to calling thread */
                 reply.type = GNRC_NETAPI_MSG_TYPE_ACK;
                 reply.content.value = (uint32_t)res;
                 msg_reply(&msg, &reply);
                 break;
             }
-            default:
-                LOG_ERROR("ERROR: [LWMAC] Unknown command %" PRIu16 "\n", msg.type);
+
+            /* RTT raised an interrupt */
+            case GNRC_LWMAC_EVENT_RTT_TYPE: {
+                if (gnrc_netdev_lwmac_get_dutycycle_active(netif)) {
+                    rtt_handler(msg.content.value, netif);
+                    lwmac_schedule_update(netif);
+                }
+                else {
+                    LOG_DEBUG("[LWMAC] Ignoring late RTT event while duty-cycling is off\n");
+                }
+                break;
+            }
+
+            /* An LWMAC timeout occurred */
+            case GNRC_LWMAC_EVENT_TIMEOUT_TYPE: {
+                gnrc_lwmac_timeout_make_expire((gnrc_lwmac_timeout_t *) msg.content.ptr);
+                lwmac_schedule_update(netif);
+                break;
+            }
+
+            default: {
+                if (netif->ops->msg_handler) {
+                    DEBUG("gnrc_netif2: delegate message of type 0x%04x to "
+                          "netif->ops->msg_handler()\n", msg.type);
+                    netif->ops->msg_handler(netif, &msg);
+                }
+#if ENABLE_DEBUG
+                else {
+                    DEBUG("gnrc_netif2: unknown message type 0x%04x"
+                          "(no message handler defined)\n", msg.type);
+                }
+#endif
                 break;
+            }
         }
 
         /* Execute main state machine because something just happend*/
-        while (gnrc_netdev_lwmac_get_reschedule(gnrc_netdev)) {
-            lwmac_update(gnrc_netdev);
+        while (gnrc_netdev_lwmac_get_reschedule(netif)) {
+            lwmac_update(netif);
```

Comment:
Mh this might be a problem... but you can propably just put this into the different functions of `gnrc_netif2_ops_t`. Yes, that is code duplication, but so is copying most of the code of `gnrc_netif2.c` ;-). And I think with the way I proposed it, this would lead to less code duplication.